index.js

```diff
@@ -1 +1,133 @@
-module.exports = require('./lib/emoji');
\ No newline at end of file
+'use strict'
+
+const charRegex = require('char-regex')()
+const emojiRegex = require('emoji-regex')()
+const randomItem = require('random-item')
+const skinTone = require('skin-tone')
+const is = require('@sindresorhus/is')
+const { assert } = is
+
+const emojiData = Object.entries(require('emojilib').lib).map(([name, { char: emoji }]) => [name, emoji])
+
+const emoji = new Map(emojiData)
+const inverted = new Map(emojiData.map(([name, emoji]) => ([emoji, name])))
+
+function normalizeName (name) {
+  if (/:.+:/.test(name)) {
+    name = name.slice(1, -1)
+  }
+
+  return name
+}
+
+function replace (input, replacement, { removeSpaces = false } = {}) {
+  assert.string(input)
+  assert.any([is.string, is.function], replacement)
+  assert.boolean(removeSpaces)
+
+  const replaceFunction = typeof replacement === 'function' ? replacement : () => replacement
+
+  const characters = input.match(charRegex)
+
+  if (characters === null) {
+    return input
+  }
+
+  return characters
+    .map((character, index) => {
+      const emoji = exports.which(character)
+
+      if (!emoji) {
+        return character
+      }
+
+      if (removeSpaces && characters[index + 1] === ' ') {
+        characters[index + 1] = ''
+      }
+
+      return replaceFunction(emoji, index, input)
+    })
+    .join('')
+}
+
+exports.get = name => {
+  assert.string(name)
+
+  return emoji.get(normalizeName(name))
+}
+
+exports.which = (emoji, { markdown = false } = {}) => {
+  assert.string(emoji)
+  assert.boolean(markdown)
+
+  const result = inverted.get(skinTone(emoji, 'none'))
+
+  if (result === undefined) {
+    return undefined
+  }
+
+  return markdown ? `:${result}:` : result
+}
+
+exports.random = () => {
+  const [name, emoji] = randomItem(emojiData)
+  return { name, emoji }
+}
+
+exports.search = keyword => {
+  assert.string(keyword)
+
+  keyword = normalizeName(keyword)
+
+  return emojiData
+    .filter(([name]) => name.includes(keyword))
+    .map(([name, emoji]) => ({ name, emoji }))
+}
+
+exports.find = emoji => {
```

Comment:
Before there were `*ByCode` and `*ByName` equivalents for APIs such as `.find`. Was it intentional to remove them in v2? (I can't find discussion)

Reply:
@omnidan thoughts?

Reply:
I would not know of a reason to remove them, maybe they were just forgotten about in the rewrite? /cc @Richienb

Reply:
Added them back in #113.